Makefile

```diff
@@ -308,11 +309,17 @@ out/doc/%: doc/%
 # check if ./node is actually set, else use user pre-installed binary
 gen-json = tools/doc/generate.js --format=json $< > $@
 out/doc/api/%.json: doc/api/%.md
+	[ -e tools/doc/node_modules/js-yaml/package.json ] || \
+		[ -e tools/eslint/node_modules/js-yaml/package.json ] || \
+		( cd tools/doc && ( $(NODE) "$(NPM)" install || node "$(NPM)" install ) )
```

Comment:
Maybe prefix with `[ -x $(NODE) ]` or wrap the whole block in a `if [ -x $(NODE) ]; then ... fi`?  That's what the surrounding code does.